inc/class-core-sitemaps-stylesheet.php

```diff
@@ -0,0 +1,156 @@
+<?php
+/**
+ * The Core_Sitemaps_Stylesheet sitemap provider.
+ *
+ * This class provides the XSL stylesheets to style all sitemaps.
+ *
+ * @package Core_Sitemaps
+ */
+
+/**
+ * Class Core_Sitemaps_Users
+ */
+class Core_Sitemaps_Stylesheet {
+	/**
+	 * Renders the xsl stylesheet.
+	 *
+	 * @return string $xsl XSL file.
+	 */
+	public function render_stylesheet() {
+		$stylesheet_query = get_query_var( 'stylesheet' );
+
+		if ( 'xsl' === $stylesheet_query ) {
+			header( 'Content-type: application/xml; charset=UTF-8' );
+
+			$xsl = $this->stylesheet_xsl();
+
+			/**
+			 * Filter the content of the sitemap stylesheet.
+			 *
+			 * @param string $xsl Full content for the xml stylesheet.
+			 */
+			return apply_filters( 'core_sitemaps_stylesheet_content', $xsl );
```

Comment:
Instead of returning here, you'll need to echo the XSL and then `exit` so WordPress doesn't continue with it's templating logic.